cmd/kueuectl/app/create/create_clusterqueue.go

```diff
@@ -0,0 +1,489 @@
+		err = o.parseResourceGroups()
+	}
+	if err != nil {
+		return err
+	}
+
+	clientset, err := clientGetter.KueueClientSet()
+	if err != nil {
+		return err
+	}
+
+	o.Client = clientset.KueueV1beta1()
+
+	o.DryRunStrategy, err = util.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+
+	err = util.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+	if err != nil {
+		return err
+	}
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.PrintObj = printer.PrintObj
+
+	return nil
+}
+
+// Validate validates required fields are set to support structured generation
+func (o *ClusterQueueOptions) Validate(ctx context.Context) error {
+	if len(o.Name) == 0 {
+		return fmt.Errorf("name must be specified")
+	}
+
+	return nil
+}
+
+// Run create clusterqueue
+func (o *ClusterQueueOptions) Run(ctx context.Context) error {
+	cq := o.createClusterQueue()
+	if o.DryRunStrategy != util.DryRunClient {
+		var (
+			createOptions metav1.CreateOptions
+			err           error
+		)
+		if o.DryRunStrategy == util.DryRunServer {
+			createOptions.DryRun = []string{metav1.DryRunAll}
+		}
+		cq, err = o.Client.ClusterQueues().Create(ctx, cq, createOptions)
+		if err != nil {
+			return err
+		}
+	}
+	return o.PrintObj(cq, o.Out)
+}
+
+func (o *ClusterQueueOptions) createClusterQueue() *v1beta1.ClusterQueue {
+	return &v1beta1.ClusterQueue{
+		TypeMeta:   metav1.TypeMeta{APIVersion: v1beta1.SchemeGroupVersion.String(), Kind: "ClusterQueue"},
+		ObjectMeta: metav1.ObjectMeta{Name: o.Name},
+		Spec: v1beta1.ClusterQueueSpec{
+			Cohort:            o.Cohort,
+			QueueingStrategy:  o.QueueingStrategy,
+			NamespaceSelector: &o.NamespaceSelector,
+			Preemption: &v1beta1.ClusterQueuePreemption{
+				ReclaimWithinCohort: o.ReclaimWithinCohort,
+				WithinClusterQueue:  o.PreemptionWithinClusterQueue,
+			},
+			ResourceGroups: o.ResourceGroups,
+		},
+	}
+}
+
+func (o *ClusterQueueOptions) parseResourceGroups() error {
+	nominalQuotas, err := parseUserSpecifiedResourceQuotas(o.UserSpecifiedNominalQuota, nominalQuota)
+	if err != nil {
+		return err
+	}
+
+	borrowingLimits, err := parseUserSpecifiedResourceQuotas(o.UserSpecifiedBorrowingLimit, borrowingLimit)
+	if err != nil {
+		return err
+	}
+
+	lendingLimits, err := parseUserSpecifiedResourceQuotas(o.UserSpecifiedLendingLimit, lendingLimit)
+	if err != nil {
+		return err
+	}
+
+	resourceGroups, err := mergeResourcesByFlavor(slices.Concat(nominalQuotas, borrowingLimits, lendingLimits))
+	if err != nil {
+		return err
+	}
+
+	resourceGroups, err = mergeFlavorsByCoveredResources(resourceGroups)
+	if err != nil {
+		return err
+	}
+
+	o.ResourceGroups = resourceGroups
+
+	return nil
+}
+
+func parseUserSpecifiedResourceQuotas(resources []string, quotaType string) ([]v1beta1.ResourceGroup, error) {
+	var resourceGroups []v1beta1.ResourceGroup
+
+	regex := regexp.MustCompile(`^(\w+):(\w+=\w+;)*\w+=\w+;?$`)
```

Comment:
Please use the same regex as kubernetes validation.

Reply:
Uhm... there isn't a single regex, but here is the starting point https://github.com/kubernetes/kubernetes/blob/5d88b0e08f26055290ca808db061efc841494f93/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L42

Reply:
Here I was thinking of doing just a basic validation of the flag format and delegating all of the value validations to the server side. It might be best to delegate the value validations to the server because otherwise, we would need to duplicate the logic from webhooks.

What do you think?

Reply:
Having a weaker validation in the CLI seems reasonable, but not too weak.

How are you matching `/`? I see a `*` in the regex above, which seems too permissive.

Reply:
nvm, the new regex looks reasonable.

Reply:
Yeah, the current regex is `(\w+):((\w+[\.-]?)*\/?\w+=\w+;)*(\w+[\.-]?)*\/?\w+=\w+;?` which matches the `/`.